packages/server/dataloader/UpdatableCacheDataLoader.ts

```diff
@@ -0,0 +1,34 @@
+import DataLoader, {CacheMap} from 'dataloader'
+
+/**
+ * Adds an ability to update specific properties of a cached item. Uses a custom cache object for that.
+ */
+class UpdatableCacheDataLoader<Key, Value, CacheKey = Key> extends DataLoader<
+  Key,
+  Value,
+  CacheKey
+> {
+  private cache: CacheMap<CacheKey, Promise<Value>>
+  private readonly cacheKeyFn: (key: Key) => CacheKey
+
+  constructor(
+    batchLoadFn: DataLoader.BatchLoadFn<Key, Value>,
+    options?: DataLoader.Options<Key, Value, CacheKey>
+  ) {
+    const cacheMap = new Map()
```

Comment:
+1 it looks like the cacheMap types could be anything. Perhaps we should add the types like `const cacheMap = new Map<CacheKey, Promise<Value>>()`?

Reply:
@nickoferrall make sense, added